android/jni/cpu_features/Android.mk

```diff
@@ -0,0 +1,8 @@
+LOCAL_PATH := $(call my-dir)/../../../prebuilt/android-$(TARGET_ARCH)/cpu_features/lib
+
+include $(CLEAR_VARS)
+LOCAL_ARM_MODE := $(MY_ARM_MODE)
+LOCAL_MODULE := cpu_features
+LOCAL_SRC_FILES := libndk_compat.so
```

Comment:
Historically we used `libcpufeatures.so` as `cpu-features` library name. Do you know if it is possible to continue using that name?

Reply:
Yes it is possible, but I don't see why the true name is worse. I mean, we import a well-known library from a quite respectable source. Renaming it may only cause confusion and result in that some people will end with two copies of the library (if they use it their NDK code).

Note that in the long run, it is possible to gradually switch to the more detailed **libcpu_features.so**, and also, probably, use it for iOS.

Reply:
The idea is simple. Name of the library is `cpu-features` and the name of the library file is `libndk_compat.so`. They look like two different things. I just expect to have a filename which will include the name of the library. So, people looking inside the .apk can understand what it is.

Reply:
OK then, you could name the library `ndk-compat`. Or hide it altogether from the list, consider it an implementation detail, as you don't have a separate entry for `mobileffmpeg_abidetect` or `androidltssupport`.

Alternatively, you can use it as a static library, then the name of `libndk_compat.a` can be harmlessly changed to `libcpufeatures.a`.

Finally, the long-term solution would be to use the modern `libcpu_features.so` that is part of the same Google repo. Let me explain.

Google developed this cross-platform library that handles more CPU features than the small one that is (still) included in NDK, but its API is different. They maintain the `libndk_compat.so` library in the same GitHub project, which is Android-only and probably less advanced, but replaces seamlessly the library that the Android developers are used to.

Reply:
`mobile-ffmpeg` can switch to main `cpu-features` library, even today. However, `libvpx` still requires NDK compatible one. So, we have to wait for that.

Generating a static library can be a solution. But, it is not the ideal choice. Because linking the same static library from multiple libraries creates complications about the symbols. I'm not % 100 sure that it will be possible to resolve them and build a running version.

Hiding the library name is possible. I'm already doing this in current version of the scripts. But this is not my main concern. My main concern is the file name. `ndk_compat` is not telling me anything about `cpu-features`.

By the way, I can solve it in my way if you don't want to do anything about it.

Reply:
I find no complications when I use **ndk_compat** as a static library. The change is in a separate commit, and can be easily reverted if you wish.

Reply:
I guess you're right about the static version of the library. Two different libraries are linked with it. So, it shouldn't be a problem.